.clang-format

```diff
@@ -2,11 +2,13 @@
 # Defaults for all languages.
 BasedOnStyle:  Google
 
-# Setting ColumnLimit to 0 so developer choices about where to break lines are maintained.
-# Developers are responsible for adhering to the 120 character maximum.
-ColumnLimit:     0
+ColumnLimit: 120
 SortIncludes: false
 DerivePointerAlignment: false
+AlignAfterOpenBracket: BlockIndent
```

Comment:
Why this? What is the benefit? From looking at a few random files it seems counter-productive by increasing the number of lines and diminishing readability (at least to me) by changing where things are aligned.

Reply:
The suggestion was https://github.com/microsoft/onnxruntime/pull/15524#discussion_r1171951117, first point. Reproducing it here, opinions welcome:

> * it's much harder to diff the addition/removal/reordering/renaming of parameters in function definitions and function calls with ragged parameter wrap. Either putting all params on the same line when they fit, or breaking one param per line when they don't is linearly scannable and trivially diffable. I suspect the 7 codebases I've worked in at Microsoft (which all do this, and all have had long lifespans) intentionally do this for easier security reviews during maintenance, because anything that makes it easier for the eyes is likely to reduce goofs for the author and help reviewers spot issues.

Reply:
*Scott*: I also wonder if Justin's clang config should be using some other combo of attributes, because it seems to have some other unintended impacts 🤨, but generally I stand by the observation that diffing and visually validating insertions/renames of parameters is *much* clearer when parameters in function definitions are arranged either horizontally or vertically, rather than a zigzag mishmash of the two (something I was guilty of in my earlier coding days too, before I cared about simpler diffs and maintenance 😅):

![image](https://user-images.githubusercontent.com/1809166/233753848-508ca925-6207-4e32-8368-4c59c8888c7f.png)

Regarding the increase in the number of lines, since we're not printing our code onto physical paper anymore, we're not saving any trees by keeping things dense. Personally (of the 3 options I show below), I may like the first the most (very easy to spot diffs and to insert/remove params, with clarity and regularity of alignment), whereas the second seems to be what ORT usually uses (but also often not), but I'm particularly opposing the third one in the image, which raggedly jams multipleparametersintoadenseline:

![image](https://user-images.githubusercontent.com/1809166/233754426-65f8192f-aeca-4937-8708-b9f3daed8e74.png)

Reply:
Only guarantee is that you will never get 100% of developers to agree on how it should look. :P 

> diffing and visually validating insertions/renames of parameters is much clearer when parameters in function definitions are arranged either horizontally or vertically

This seems slightly contradictory. You're ok with diffing if they're all on one line but not if the params are split over multiple lines and there's more than one param on a line?

I would vote for everything on one line if it fits. 
If it doesn't, I would not vote for 'make each line as long as possible' as I think that's harder to parse. 
I'm fine with one line per param, but that can get a little excessive in a few places where a function has a lot of args (not common but does happen). I'd rather have the ability to choose the line breaks so I can group related params on a single line to reduce the number of lines, but understand that's impossible to enforce and some folks idea of what 'reasonable' grouping is could be completely different to mine. (e.g. for AddElementWiseIf in the example I'd group the first 3 input 'index' args on one line, and have the output args on the next).

I strongly prefer the indent using the opening '(' as I don't like clutter on the left. It slows down skimming through the code to find meaningful lines. If you need to parse the args (less common) they're still aligned, but I can go to the actual code (more common) if the args are further out of the way as there's less thinking required to skip over the bigger chunk of whitespace.

Same reason I strongly prefer an empty line after the closing '}' of the end of a 'while' or 'else'. That space lets me skip to the next chunk of code more easily.

I'm not against extra lines, but that definitely doesn't seem to be universal. Any arguments that involve printing should automatically be excluded IMHO. :) 

FWIW I would find your last example with multiple params per line far easier to parse if there was whitespace between each declaration. Essentially that's what's happening in your first example by putting the closing ')' on a different line so I wonder whether that is the more important aspect than the amount of indentation.

Reply:
> functions with many arguments 

I can think of an upside of making them ugly. When people cannot bear how it looks, it will be natural for a refactor, which is usually what these functions need.